### ISSUE TITLE
Option to exclude shipping costs

### ISSUE THREAD

**Issue by TomDeSmet** [verbb/gift-voucher#63]
**What are you trying to do?**
As the title says. Could be an interesting option.

**Comment by engram-design** [comment]
For clarity, do you mean for the redemption of a gift voucher to also include or exclude the shipping costs in the discount?

Or do you mean when purchasing a gift voucher?

**Comment by TomDeSmet** [comment]
I mean for the redemption of a gift voucher.

**Comment by engram-design** [comment]
It should be doing that currently, see the attached example 
![image](https://user-images.githubusercontent.com/1221575/111773308-20e8ce00-8902-11eb-86e2-66f48a1c2259.png)

**Comment by TomDeSmet** [comment]
Yes, but my suggestion is to have an option to exclude the shipping cost, so that you always have to pay that and the voucher only discounts the product cost.

Much like you have a setting "beforeTax" we could have a setting "beforeShippingCosts".

**Comment by engram-design** [comment]
Ah, of course, my mistake.

**Comment by sjcallender** [comment]
@engram-design - This has come up for us on a current project. Any way to convince you to have a look at adding it soon? I think we'll have to roll a custom route until then as we missed this detail previously in our project.

**Comment by engram-design** [comment]
Just added a `includeShipping` plugin setting, which is `true` by default. With this, voucher redemption values will either include or exclude shipping costs.

So you might have a $20 line item, $10 shipping, and a $50 voucher. 

- `includeShipping = true`, your order total will be $0, with $20 remaining on your voucher
- `includeShipping = false`, your order total will be $10, with $30 remaining on your voucher

To get this early, run `composer require verbb/gift-voucher:"dev-craft-4 as 3.0.17"` or `composer require verbb/gift-voucher:"dev-craft-5 as 4.0.7"`.

**Comment by sjcallender** [comment]
That works perfectly. Thanks!!

**Comment by sjcallender** [comment]
@engram-design - Actually, when using weight rate shipping rules per line item, the voucher is paying for the shipping rate still. This only prevents paying for order level shipping. Would it be possible to have it apply to line item shipping rates too?

**Comment by engram-design** [comment]
Ah right, I'll take a look at what's involved there, that might be a little tricky...

**Comment by sjcallender** [comment]
Thanks. We're trying to go live by Wednesday, so if you can't or decide not to, would you let me know so we can tackle something on our end?

**Comment by engram-design** [comment]
Updated for the next release. To get this early, run `composer require verbb/gift-voucher:"dev-craft-4 as 3.0.17"` or `composer require verbb/gift-voucher:"dev-craft-5 as 4.0.7"`.

**Comment by sjcallender** [comment]
@engram-design - What are your thoughts on supporting the exclusion of line item shipping costs?
